requester/print.go

```diff
@@ -45,53 +48,71 @@ type report struct {
 	delayLats []float64
 
 	results chan *result
+	done    chan bool
 	total   time.Duration
 
 	errorDist      map[string]int
 	statusCodeDist map[int]int
 	lats           []float64
 	sizeTotal      int64
-
-	output string
+	numRes         int64
+	output         string
 
 	w io.Writer
 }
 
-func newReport(w io.Writer, size int, results chan *result, output string, total time.Duration) *report {
+func newReport(w io.Writer, results chan *result, output string, n int) *report {
+	cap := min(n, maxRes)
 	return &report{
 		output:         output,
 		results:        results,
-		total:          total,
+		done:           make(chan bool, 1),
 		statusCodeDist: make(map[int]int),
 		errorDist:      make(map[string]int),
 		w:              w,
+		connLats:       make([]float64, 0, cap),
+		dnsLats:        make([]float64, 0, cap),
+		reqLats:        make([]float64, 0, cap),
+		resLats:        make([]float64, 0, cap),
+		delayLats:      make([]float64, 0, cap),
+		lats:           make([]float64, 0, cap),
 	}
 }
 
-func (r *report) finalize() {
+func runReporter(r *report) {
+	// Loop will continue until channel is closed
 	for res := range r.results {
+		r.numRes++
 		if res.err != nil {
 			r.errorDist[res.err.Error()]++
 		} else {
-			r.lats = append(r.lats, res.duration.Seconds())
 			r.avgTotal += res.duration.Seconds()
 			r.avgConn += res.connDuration.Seconds()
 			r.avgDelay += res.delayDuration.Seconds()
 			r.avgDNS += res.dnsDuration.Seconds()
 			r.avgReq += res.reqDuration.Seconds()
 			r.avgRes += res.resDuration.Seconds()
-			r.connLats = append(r.connLats, res.connDuration.Seconds())
-			r.dnsLats = append(r.dnsLats, res.dnsDuration.Seconds())
-			r.reqLats = append(r.reqLats, res.reqDuration.Seconds())
-			r.delayLats = append(r.delayLats, res.delayDuration.Seconds())
-			r.resLats = append(r.resLats, res.resDuration.Seconds())
+			if len(r.resLats) < maxRes {
```

Comment:
Shouldn't you report that you are actually dropping data?

Reply:
I already do, see below
```
		if r.numRes > maxRes {
			r.printf("\nNote:  Distributions are for first %d results.", len(r.lats))
		}
```